### ISSUE TITLE
[API] Math Improvements

### ISSUE THREAD

**Issue by Alan-FGR** [AtomicGameEngine/AtomicGameEngine#1638]
# TO-DO
- [ ] Clamp Magnitude
- [ ] Use a better approach for Vector Operations
- [ ] Expose/port the entire Math API to Scripting
- [ ] Easy Eulerian angles rotation API
- [ ] 2D Direction Setter

# Discussion and Details

## Clamp Magnitude
A way to clamp the magnitude of a vector is generally useful, especially when dealing with non normalized input, here's my quick implementation in C# (extension):
```C#
public static Vector2 ClampMagnitude(this Vector2 v, float maxMagnitude)
    {
        if (v.Length <= maxMagnitude)
            return v;
        return Vector2.Normalize(v) * maxMagnitude;
    }
```

## Vector Operations
It's somewhat confusing the fact that something like `MyVector3.Normalize()` is valid but won't work, and apparently the correct usage is `Vector3.Normalize(MyVector3)`... imho we should provide instance methods for these unary operations

## Expose/port the entire Math API to Scripting
We don't have the API to raycast AABBs and some types are lacking too

## Easy Eulerian angles rotation API
Provide a way to rotate and set rotation that takes Eulerian angles, ~~and also a way to easily build a Quaternion from them~~

## 2D Direction Setter
basically:
```C#
public static void SetDirection2D(this Node n, Vector2 direction)
{
    n.Rotation2D = (float)(Math.Atan2(direction.X, -direction.Y) / Math.PI) * 180;
}
```

**Comment by mattbenic** [comment]
@Alan-FGR is this still in progress?

**Comment by Alan-FGR** [comment]
Nope :trollface:... but only important task is #3 which is covered somewhere else, other stuff is minimal.
